### PR TITLE
Database

### LINKED ISSUE #20: Create a database
Set up database for storing information of the users and their reports:

- [x] Development
- [x] Production

### PR DESCRIPTION
**Pull request by hoslack**
- Created an express server and added mongoose to access an online  database 
-  Made a route for reports and controller to add report to the database
Closes #20

### CODE REVIEW

**Comment by Karyum** [comment]
love this priority 3 label on the PR :laughing: it's like saying oh it's not really important to review this right now :rofl:

**Review by Karyum — CHANGES_REQUESTED** [review]
ohh no i done screwed up...
I should have noticed that you have more than 1 `package.json` before....

Cause you don't need 3 of them to run the server react and the app itself, 1 is enough it's just gonna be a mess of a file structure loads of `package-lock.json` and very confusing

apologies for not catching this before, and although we might get merge conflicts i need you guys to move everything ( server, react ) into 1 main `package.json` in the root directory of you project.

**Comment by Karyum** [comment]
@hoslack  check my review then assign @MynahMarie @MarlenAw to check it out and then reassign me

**Comment by hoslack** [comment]
@Karyum Please merge the first PR first before I make my changes, is that possible?

**Comment by Karyum** [comment]
@hoslack yeah but why do you want me to merge this one ?
the best case senario here it to merge @MynahMarie PR #21 and then you would pull the new master and do the changes. don't worry i'll be here as soon as you make the new PR to check this. cause this PR will block your project progress, make sure that @MynahMarie @MarlenAw review the PR cause this is a big change

**Comment by hoslack** [comment]
@Karyum  Kindly let us do the refactoring in the next PR, the conflict that's coming ahead is like lightning!

**Comment by Karyum** [comment]
@hoslack 
we could avoid this conflict by doing what i told you, wait untill #21 gets merged and then do the changes on your branch and barabing barabong, if there is a better way you think this can happen then please explain it.

**Comment by hoslack** [comment]
Please help me understand what I should do, maybe I don't get it. Thanks

**Comment by Karyum** [comment]
as i said [here](https://github.com/FACN3/easyRep/pull/22#issuecomment-364741203) wait until #21 gets merged which from the looks of it gonna be soon its just a matter of commiting into the branch and then pull the new master and do the change that i requested

was that clear?

**Comment by hoslack** [comment]
That's crystal clear. I just have to wait. Great!

**Comment by hoslack** [comment]
Hello, @Karyum Allow us to use the 2 package.json files in development and later we will run a build script to minimize what we put in production. That's the easiest way to use create-react-app. We have tried using one package file but it still asking for the other one inside the client!

**Comment by Karyum** [comment]
i don't understand at what point create-react-app tell you, that you need another package.json so you get it to work, the few times i have used create react app i didn't have to do this, currently if i want to use a project to learn how to use create react app  i would use @matthewdking 's one https://github.com/facn2/path-18-app and he has only 1 package.json there.

But if you insist that you want to do it this way, and your not sure how to do it other ways then i'll be waiting for @MynahMarie and @MarlenAw 's approval on this so i can start code reviewing

**Review by MarlenAw — APPROVED** [review]

**Review by MarlenAw — APPROVED** [review]

File: models/Reports.js
Comment (MynahMarie) [inline models/Reports.js]: Please correct the little typo there... (reportsShema should be reportsSchema)

**Comment by hoslack** [comment]
Thanks Mario or @Karyum . We are not Gurus yet but we are heading there.

File: routes/reportRoutes.js
Comment (MynahMarie) [inline routes/reportRoutes.js]: Would it be ok to change the endpoint to something like 'submit' instead of 'create'? because we will only send the report to the database if the user actually reviews and sends it right? (correct me if I'm missing something...)

File: app.js
Comment (MynahMarie) [inline app.js]: Can we add a URL variable instead of hardcoding localhost? (Im just thinking of when we will deploy... I know its far ahead but it's such a small detail, maybe we can do it now?) I know its just a tiny detail...

**Review by MynahMarie — CHANGES_REQUESTED** [review]
Just little details @hoslack otherwise it looks really good!

File: routes/reportRoutes.js
Comment (hoslack) [reply routes/reportRoutes.js]: @MynahMarie, I think submit is more for a form in the front end, create means making a new collection in the database. This is for simple understanding for someone who might want to use the api even for a different project

File: app.js
Comment (hoslack) [reply app.js]: Done @MynahMarie

File: routes/reportRoutes.js
Comment (MynahMarie) [reply routes/reportRoutes.js]: @hoslack ok then!

**Review by MynahMarie — APPROVED** [review]

File: app.js
Comment (Karyum) [inline app.js]: Hmm don't you want to run your server when you database actually connected ?
so `mongoose.connect` can have event listeners, for example on error and on connected.

check [this out](http://theholmesoffice.com/mongoose-connection-best-practice/#managingthemongooseconnection)
so i would highly suggest to add the error event listener and the `connected` listener for when the database has connected and then you would want your server to connect after the db

File: routes/reportRoutes.js
Comment (Karyum) [reply routes/reportRoutes.js]: so this is `/create` but what is sit creating so i would say add to the endpoint the thing you are creating cause it is not obvious

**Review by Karyum — CHANGES_REQUESTED** [review]
Schemas might need more validation but i guess this is fine for now. :shipit:

File: app.js
Comment (hoslack) [reply app.js]: Great, done!!

**Comment by hoslack** [comment]
@Karyum Already changed. Great changes!

**Review by Karyum — APPROVED** [review]

**Pull request merged**
